src/Serilog/Settings/KeyValuePairs/KeyValuePairSettings.cs

```diff
@@ -135,12 +159,63 @@ select new
                         .GroupBy(call => call.MethodName)
                         .ToList();
 
-                    ApplyDirectives(calls, methods, CallableDirectiveReceivers[receiverGroup.Key](loggerConfiguration));
+                    ApplyDirectives(calls, methods, CallableDirectiveReceivers[receiverGroup.Key](loggerConfiguration), declaredLevelSwitches);
                 }
             }
         }
 
-        static void ApplyDirectives(List<IGrouping<string, ConfigurationMethodCall>> directives, IList<MethodInfo> configurationMethods, object loggerConfigMethod)
+        static IReadOnlyDictionary<string, LoggingLevelSwitch> ParseNamedLevelSwitchDeclarationDirectives(Dictionary<string, string> directives)
+        {
+            var matchLevelSwitchDeclarations = new Regex(LevelSwitchDeclarationDirectiveRegex);
+
+            var switchDeclarationDirectives = (from wt in directives
+                                               where matchLevelSwitchDeclarations.IsMatch(wt.Key)
+                                               let match = matchLevelSwitchDeclarations.Match(wt.Key)
+                                               select new
+                                               {
+                                                   SwitchName = match.Groups["switchName"].Value,
+                                                   InitialSwitchLevel = wt.Value
+                                               }).ToList();
+
+            var namedSwitches = new Dictionary<string, LoggingLevelSwitch>();
+            foreach (var switchDeclarationDirective in switchDeclarationDirectives)
+            {
+                LoggingLevelSwitch newSwitch;
+                if (string.IsNullOrEmpty(switchDeclarationDirective.InitialSwitchLevel))
+                {
+                    newSwitch = new LoggingLevelSwitch();
+                }
+                else
+                {
+                    var initialLevel = (LogEventLevel)SettingValueConversions.ConvertToType(switchDeclarationDirective.InitialSwitchLevel, typeof(LogEventLevel));
+                    newSwitch = new LoggingLevelSwitch(initialLevel);
+                }
+                namedSwitches[switchDeclarationDirective.SwitchName] = newSwitch;
+
+            }
+            return new ReadOnlyDictionary<string, LoggingLevelSwitch>(namedSwitches);
+        }
+
+        static LoggingLevelSwitch LookUpSwitchByNameOrThrow(string minimumLevelControlledByLevelSwitchName, IReadOnlyDictionary<string, LoggingLevelSwitch> declaredLevelSwitches)
```

Comment:
Could we just call the argument `switchName`? The result might not always be used in a `minimum-level:controlled-by` directive, so the name might be a bit too specific.

Reply:
yep :) That's a leftover of R# *Extract Method* ;)